features/step_definitions/repository_steps.rb

```diff
@@ -163,11 +167,19 @@
 end
 
 When(/^I push some commits to the repository$/) do
-  post repository_notify_push_path(id: @repository.id), {}, {'HTTP_X_GITLAB_EVENT' => 'Push Hook'}
+  request = FactoryGirl.build(:gitlab_webhook_request)
+  request.headers.each do |k, v|
+    header k, v
```

Comment:
There is a header values handler at https://softwarepublico.gov.br/gitlab/mezuro/prezento/blob/colab/features/support/header.rb. What do tou think about making it mainstream and using it here?